python-main.js

```diff
@@ -916,6 +916,39 @@ function web_editor(config) {
         return fullHex;
     }
 
+    // Trap focus in modal and pass focus to first actionable element
+    function focusModal() {
+        document.querySelector('body > :not(.vex)').setAttribute('aria-hidden', true);
+        var dialog = document.querySelector('.modal-div');
+        var focusableEls = dialog.querySelectorAll('a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled])');
+        $(focusableEls).each(function() {
+            $(this).attr('tabindex', '0');
+        });
+        dialog.focus();
+        dialog.onkeydown = function(event) {
+            if (event.which == 9) {
+                // if tab key is pressed
+                var focusedEl = document.activeElement;
+                var numberOfFocusableEls = focusableEls.length;
+                var focusedElIndex = Array.prototype.indexOf.call(focusableEls, focusedEl);
+
+                if (event.shiftKey) {
```

Comment:
This doesn't look compatible with IE10 or Safarit <= 12.
https://caniuse.com/#feat=mdn-api_keyboardevent_shiftkey